### PR TITLE
[EPPETA-65] User Creation flow updates

### PR DESCRIPTION
**Pull request by jleiva-gap**
- Update create account, so it does not add a default role to new users, except if the flag to create users as adminstrator is true
- Update Account control to add or remove roles to an user
- New Users are created without role
- Enable navigation menu
- Update Authorization by role
  - Menu options
  - Pages
- Add an user list for administrator
- Add a page to update user roles.
- Add support for multiple roles by user
- Update initial page:
  - Mentor and Supervisor: Evaluation List.
  - Administrator: User list.
  - User without role: Access denied message

### CODE REVIEW

File: src/reactapp/src/components/Navbar.js [hunk 1]
```diff
@@ -290,44 +291,51 @@ const MobileNavItem = ({ label, children, href }) => { // Remove type annotation
...
+    children: [
+      {
+        label: 'Evaluations',
+        subLabel: 'List of evaluations',
+        href: '/main'
+      }
+    ],
+  },
+  {
+    label: 'User',
```
Comment (stevenarnold-ed-fi) [inline src/reactapp/src/components/Navbar.js]: 'Administration' instead of 'User'

File: src/reactapp/src/components/Navbar.js [hunk 2]
```diff
@@ -290,44 +291,51 @@ const MobileNavItem = ({ label, children, href }) => { // Remove type annotation
...
-    //            href: '#',
-    //        },
-    //    ],
-    //},
-    //{
-    //    label: 'Teacher Preparation Program',
-    //    href: '#',
-    //},
+  {
+    label: 'Evaluation',
```
Comment (stevenarnold-ed-fi) [inline src/reactapp/src/components/Navbar.js]: Can we hide or not have this section. With adding a way to get back to the main page in ETA-79, I don't think this as a menu item is desired.

File: src/reactapp/src/components/AuthenticatedRoute.js [hunk 3]
```diff
@@ -32,10 +32,17 @@ const AuthenticatedRoute = ({ element: authenticatedComponent, ...rest }) => {
         navigate("/login");
         sessionStorage.clear();
       }
-    }
-    );
+      if (roles && !roles.includes('*') && !isLoggedInUserInRole(roles)) {
+        setAlertMessageText("You do not have access to the requested page");
```
Comment (stevenarnold-ed-fi) [inline src/reactapp/src/components/AuthenticatedRoute.js]: Please replace with "You do not have access to the requested page, contact your administrator if you think this is in error."

File: src/reactapp/src/components/Navbar.js [hunk 4]
```diff
@@ -290,44 +291,51 @@ const MobileNavItem = ({ label, children, href }) => { // Remove type annotation
 // Disable interface, interface are available for Typescript
 interface NavItem {
     label: string;
+    roles?: Array<string>
     subLabel?: string;
     children?: Array<NavItem>;
     href?: string;
 }*/
+const showMenuByRol = () => {
```
Comment (stevenarnold-ed-fi) [inline src/reactapp/src/components/Navbar.js]: rename to showMenuByRole

**Review by stevenarnold-ed-fi — CHANGES_REQUESTED** [review]

**Comment by stevenarnold-ed-fi** [comment]
@jleiva-gap Forgot to add: This doesn't seem to cause any usability issues, but the User Administration page is throwing errors

File: src/reactapp/src/components/Navbar.js (hunk 2 above)
Comment (jleiva-gap) [reply src/reactapp/src/components/Navbar.js]: I added a redirect to the home page using the image, an I am working on the complete solution in EPPETA-79, but we have a way to get back.

**Review by stevenarnold-ed-fi — APPROVED** [review]

**Pull request merged**